app/adapters/signal_adapter/outbound/text.rb

```diff
@@ -36,6 +29,32 @@ def data
           message: text
         }
       end
+
+      def handle_response(response)
+        case response.code.to_i
+        when 200
+          # TODO: Do something on success. For example, mark the message as delivered?
+          # Or should we use deliver receipts as the source of truth.
+          Rails.logger.debug 'Great!'
+        when 400..599
+          error_message = JSON.parse(response.body)['error']
+          exception = SignalAdapter::BadRequestError.new(error_code: response.code, message: error_message)
+          context = {
+            code: response.code,
+            message: response.message,
+            headers: response.to_hash,
+            body: error_message
+          }
+          tags = if error_message.match?(/with challenge token /)
+                   regex = /"(.*?)"/
+                   challenge_token = error_message.split('with challenge token ').last.match(regex).to_s.split(/"/).last
+                   { challenge_token: challenge_token }
+                 else
+                   {}
+                 end
+          ErrorNotifier.report(exception, context: context, tags: tags)
```

Comment:
I would include a default here for every other response code.

Reply:
To be honest, as I wrote to you in Slack, I think this is overkill and makes it harder to maintain.
The error message now, since you added the `stdout` to the output can contain several different clues as to why the message failed to send and I don't like the idea of parsing the error messages for several different reasons and doing different things based on that information.

For example, we now have the full message that a number is not registered with Signal any longer if a contributor unregistered their number.

I think it is enough to send the entire message to sentry, as I did:

```
error_message = JSON.parse(response.body)['error']
exception = SignalAdapter::BadRequestError.new(error_code: response.code, message: error_message)
```

and continue sending the errors to the #monitoring channel. This channel is for actionable errors that a dev should look into/do something about.

We have another channel #support for non-actionable errors that might require some support from a non-dev.